.github/workflows/build-system-test-react.yml

```diff
@@ -0,0 +1,203 @@
+          # - framework: react
+          #   framework-version: latest
+          #   build-tool: gatsby
+          #   build-tool-version: latest
+          #   pkg-manager: npm
+          #   language: ts
+          #   node-version: 18
+          - framework: react
+            framework-version: latest
+            build-tool: vite
+            build-tool-version: latest
+            pkg-manager: npm
+            language: ts
+            node-version: 18
+          - framework: react
+            framework-version: latest
+            build-tool: vite
+            build-tool-version: 2
+            pkg-manager: npm
+            language: ts
+            node-version: 18
+
+    env:
+      MEGA_APP_NAME: ${{ matrix.framework }}-${{ matrix.framework-version }}-${{ matrix.build-tool }}-${{ matrix.build-tool-version }}-node-${{ matrix.node-version }}-${{ matrix.language }}
+      DEPENDENCIES: react@${{ matrix.framework-version }} react-dom@${{ matrix.framework-version }} @aws-amplify/ui-react aws-amplify @aws-amplify/ui-react-storage
+      DEP_TYPES: '@types/react@${{ matrix.framework-version }} @types/react-dom@${{ matrix.framework-version }}'
+
+    steps:
+      - name: Checkout Amplify UI
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3
+        with:
+          persist-credentials: false
+      - name: Setup Node.js ${{ matrix.node-version }} with ${{ matrix.pkg-manager }}
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: ${{ matrix.pkg-manager }}
+        env:
+          SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2
+      - name: Install build-system-tests package
+        uses: ./.github/actions/install-with-retries
+        with:
+          working-directory: ./build-system-tests
+      - name: Create MegaApp ${{ matrix.framework }}-${{ matrix.framework-version }}-${{ matrix.build-tool }}-${{ matrix.build-tool-version }}-node-${{ matrix.node-version }}-${{ matrix.language }}
+        run: |
+          mkdir -p mega-apps/
+          if ${{ matrix.build-tool == 'cra' && matrix.language == 'js' }};then
+            npx create-react-app mega-apps/$MEGA_APP_NAME
+          fi
+
+          if ${{ matrix.build-tool == 'cra' && matrix.language == 'ts' }};then
+            npx create-react-app mega-apps/$MEGA_APP_NAME --template typescript
+          fi
+
+          if ${{ matrix.build-tool == 'next' }};then
+            npx create-next-app mega-apps/$MEGA_APP_NAME --ts --no-src-dir --no-experimental-app --no-eslint
+          fi
+
+          if ${{ matrix.build-tool == 'gatsby' }};then
+            cd mega-apps
+            npx create-gatsby@latest -y $MEGA_APP_NAME
+            cd ..
+          fi
+
+          if ${{ matrix.build-tool == 'vite' }};then
+            cd mega-apps
+            npm create vite@${{ matrix.build-tool-version }} $MEGA_APP_NAME -- --template react-ts
+            cd ..
+          fi
```

Comment:
These scripts are getting fairly long, and I think it would be better if we write a separate shell file instead of inline scripts. That way, we get the benefit of code highlighting, formatter, and code readability for future readers.

Example: https://github.com/aws-amplify/amplify-ui/blob/main/environments/pull-environments.sh

Reply:
hahah! I was waiting for this comment. 

Done 5f8f02b6c9a2ad058713c21b5d423f75ed0d6ea0

Reply:
Changed the shell-formatter df2f0578fa681a0d9e95e8a866dcdd3a503812ca